t/16-utils-runcmd.t

```diff
@@ -296,4 +297,79 @@ subtest 'saving needle via Git' => sub {
     #       it is tested here explicitly.
 };
 
+subtest 'signal guard aborts when git is disabled and do_cleanup is no' => sub {
+    my $save_mock = Test::MockModule->new('OpenQA::Task::Needle::Save');
+    my $signal_guard = OpenQA::Task::SignalGuard->new(bless({} => 'Test::FakeMinionJob'));
+    $save_mock->redefine(_create_signal_guard => sub { return $signal_guard; });
+
+    my $git_mock = Test::MockModule->new('OpenQA::Git');
+    $git_mock->redefine(enabled => sub { 0; });
```

Comment:
Please see line 151 ff. on how to change the config in tests.
This works but I find it a bit unusual.